PWGDQ/Tasks/filterPP.cxx

```diff
@@ -258,37 +354,96 @@ struct DQFilterPPTask {
+    }
+    for (int i = 0; i < fNMuonCuts; i++) {
+      histNames += Form("TrackMuons_%s;PairsMuonsPM_%s;PairsMuonsPP_%s;PairsMuonsMM_%s;", fMuonCutsNameArray->At(i)->GetName(), fMuonCutsNameArray->At(i)->GetName(), fMuonCutsNameArray->At(i)->GetName(), fMuonCutsNameArray->At(i)->GetName());
     }
     DefineHistograms(fHistMan, histNames.Data());    // define all histograms
     VarManager::SetUseVars(fHistMan->GetUsedVars()); // provide the list of required variables so that VarManager knows what to fill
     fOutputList.setObject(fHistMan->GetMainHistogramList());
 
     // configure the stats histogram
-    fStats.setObject(new TH2I("stats", "stats", fNPairCuts, -0.5, -0.5 + fNPairCuts, fNTrackCuts, -0.5, -0.5 + fNTrackCuts));
+    fStatsSingleBarrel.setObject(new TH1I("statsSingleBarrel", "statsSingleBarrel", fNTrackCuts, -0.5, -0.5 + fNTrackCuts));
+    for (int i = 0; i < fNTrackCuts; ++i) {
+      fStatsSingleBarrel->GetXaxis()->SetBinLabel(i + 1, fTrkCutsNameArray->At(i)->GetName());
+    }
+
+    fStatsSingleMuon.setObject(new TH1I("statsSingleMuon", "statsSingleMuon", fNMuonCuts, -0.5, -0.5 + fNMuonCuts));
+    for (int i = 0; i < fNMuonCuts; ++i) {
+      fStatsSingleMuon->GetXaxis()->SetBinLabel(i + 1, fMuonCutsNameArray->At(i)->GetName());
+    }
+
+    fStatsBarrel.setObject(new TH2I("statsBarrel", "statsBarrel", fNPairCuts, -0.5, -0.5 + fNPairCuts, fNTrackCuts, -0.5, -0.5 + fNTrackCuts));
     for (int i = 0; i < fNPairCuts; ++i) {
-      fStats->GetXaxis()->SetBinLabel(i + 1, fPairCuts[i].GetName());
+      fStatsBarrel->GetXaxis()->SetBinLabel(i + 1, fPairCuts[i].GetName());
     }
     for (int i = 0; i < fNTrackCuts; ++i) {
-      fStats->GetYaxis()->SetBinLabel(i + 1, fTrkCutsNameArray->At(i)->GetName());
+      fStatsBarrel->GetYaxis()->SetBinLabel(i + 1, fTrkCutsNameArray->At(i)->GetName());
+    }
+
+    fStatsMuon.setObject(new TH2I("statsMuon", "statsMuon", fNPairCuts, -0.5, -0.5 + fNPairCuts, fNMuonCuts, -0.5, -0.5 + fNMuonCuts));
+    for (int i = 0; i < fNPairCuts; ++i) {
+      fStatsMuon->GetXaxis()->SetBinLabel(i + 1, fPairCuts[i].GetName());
+    }
+    for (int i = 0; i < fNMuonCuts; ++i) {
+      fStatsMuon->GetYaxis()->SetBinLabel(i + 1, fMuonCutsNameArray->At(i)->GetName());
     }
   }
 
-  void process(MyEventsSelected::iterator const& event, MyBarrelTracksSelected const& tracks, aod::BCs const& bcs)
+  void
+    process(MyEventsSelected::iterator const& event, MyBarrelTracksSelected const& tracks, MyMuonsSelected const& muons, aod::BCs const& bcs)
   {
-    uint64_t filter = 0;
-    constexpr int pairType = VarManager::kJpsiToEE;
+    uint64_t filter = 0; // first 32 bits for dielectrin, last 32 for dimuons
+
+    //[0] = SingleE, [1] = SingleMulow, [2] = SingleMuHigh, [3] = DiElectron, [4] = DiMuon
+    bool keepEvent[kNTriggersDQ]{false};
 
+    constexpr int pairTypeEE = VarManager::kJpsiToEE;
+    constexpr int pairTypeMuMu = VarManager::kJpsiToMuMu;
     if (event.isDQEventSelected() == 1) {
```

Comment:
maybe here use the opposite logic, if the event is not selected, then fill the event filter with zero and then exit the function